INSTALL.md

```diff
@@ -54,17 +54,51 @@ git submodule update --init --recursive -- test/googletest
 git submodule update --init --recursive -- extern/pybind11
 ```
 
-**Build NGEN using cmake:**
+**Configure NGen using cmake:**
+
+In addition to normal CMake options, the following `ngen` configuration options are available:
+
+Option                | Description
+--------------------- | -----------
+NETCDF_ACTIVE         | Include NetCDF support
+NGEN_WITH_SQLITE3     | Include SQLite3 support (GeoPackage support)
+UDUNITS_ACTIVE        | Include UDUNITS support
+MPI_ACTIVE            | Include MPI (Parallel Execuation) support
+BMI_FORTRAN_ACTIVE    | Include Fortran BMI support
+BMI_C_LIB_ACTIVE      | Include C BMI support
+NGEN_ACTIVATE_PYTHON  | Include Python support
+NGEN_ACTIVATE_ROUTING | Include `t-route` integration
+
+> These can be included by adding `-D{Option}=[ON/OFF]` to the following CMake command.
+
+> **Note**: if `-DNGEN_ACTIVATE_PYTHON=ON`, and you are using an *activated* virtual environment, then CMake will bind its Python configuration to your virtual environment. This means, when you run the `ngen` executable, your virtual environment must be activated.
+
+The following CMake command will configure the build:
+
+```shell
+cmake -DCMAKE_CXX_COMPILER=/usr/bin/g++ \
+      -DBOOST_ROOT=boost_1_72_0 \
+      -B /build \
+      -S .
+```
+
+**Build NGen using cmake:**
+
+```shell
+cmake --build build --target ngen
+```
+
+If you want to build with multiple jobs (in parallel), then you can run the following instead:
 
 ```shell
-cmake -B /ngen -S . &&
-cmake --build . --target ngen
+# replace 2 with number of jobs
+cmake --build build --target ngen -- -j 2
 ```
 
 **Running the Model:**
 
 ```shell
-./ngen data/catchment_data.geojson "" data/nexus_data.geojson "" data/example_realization_config.json
+./build/ngen data/catchment_data.geojson "" data/nexus_data.geojson "" data/example_realization_config.json
```

Comment:
```suggestion
./build/ngen data/catchment_data.geojson all data/nexus_data.geojson all data/example_realization_config.json
```
Bonus, but if we're editing this, recommend we use the more clear `all` parameter value instead of `""`.